src/linker/Linker/Annotations.cs

```diff
@@ -190,9 +194,20 @@ public void Mark (CustomAttribute attribute, in DependencyInfo reason)
 			Tracer.AddDirectDependency (attribute, reason, marked: true);
 		}
 
+		public IEnumerable<IMetadataTokenProvider> MarkedPending ()
+		{
+			foreach (var pending in marked_pending.ToArray ())
```

Comment:
Why does this need extra buffer?

Reply:
This is because marked_pending could change as we iterate over it, in general. I could make it a linked list + hashset instead, but then every Mark will need to be added to both. The common case is for MarkStep to call Mark, and CheckProcessed shortly after, so marked_pending tends to be small by the time we need it (usually empty except for the Initialize case).

Reply:
Then just return the buffer array

Reply:
Fixed to return array